### ISSUE TITLE
Closing dialog causes other components flicker on iOS Safari

### ISSUE THREAD

**Issue by septem1997** [ariakit/ariakit#3841]
### Current behavior

https://github.com/ariakit/ariakit/assets/13849245/acf39133-55d3-435c-b230-54a15a4a47cc

Note the "Examples" at the bottom of the screen, flickers when I close dialog

only happens on iPhone, works fine on desktop and Android

### Steps to reproduce the bug

1. Open https://ariakit.org/components/dialog on iphone Safari
2. Open dialog and close

### Expected behavior

No flickering

**Comment by diegohaz** [comment]
What's the iOS version? I can't see this on iOS 17.4.1.

**Comment by septem1997** [comment]
> What's the iOS version? I can't see this on iOS 17.4.1.

Sorry for the confusion. the screen recording of iPhone is from XCode simulator, I did not see this issue when accessing the Ariakit examples page on my iPhone XR, but it did happen on my website.

This is another video, using my iPhone XR to access my website (https://globus-china.com/) and open/close the dialog component, I'm sorry I can't provide a minimal demo.


https://github.com/ariakit/ariakit/assets/13849245/bded88e6-8b56-4359-953d-b54fa6e7704c



I guess this problem is more easy to occur under these two conditions：
1. Complex page rendering
2. Scroll away from the top of the page

Another thing worth mentioning is that if I set **preventBodyScroll** to false, the flickering completely disappears.

Thanks for your reply!
